src/microbe_stage/Microbe.cs

```diff
@@ -1209,6 +1212,14 @@ private void HandleCompoundAbsorbing(float delta)
 
         cloudSystem.AbsorbCompounds(Translation, grabRadius, Compounds,
             TotalAbsorbedCompounds, delta, Membrane.Type.ResourceAbsorptionFactor);
+
+        // return if both InfCompounds and PlayerMicrobe enabled
+        if (!(CheatManager.InfiniteCompounds && IsPlayerMicrobe))
```

Comment:
This still looks the same, I meant code like:

```
if(IsPlayerMicrobe && CheatManager.InfiniteCompounds)
{
    compound adding code here
}
```
which will be much more readable: `if should have infinite compounds, add them` instead of trying to go the other way around. I'm not convinced that the approach of early return really works here as this isn't like a continuation of the method, instead it's just some extra functionality tacked in here, so I think an early return actually decreases readability here.